js/client.js

```diff
@@ -1011,6 +1021,33 @@
 				break;
 			}
 		},
+		parseGroups: function (groupsList) {
+			var data = null;
+			try {
+				data = JSON.parse(groupsList);
+			} catch (e) {}
+			if (!data) return; // broken JSON - keep default ranks
+
+			Config.groups = {};
+			// process the data and sort into the three auth tiers, 0, 1, and 2
+			for (var i = 0; i < data.length; i++) {
+				var entry = data[i];
+				if (!entry) continue;
+
+				var symbol = entry.symbol || ' ';
+				var groupName = entry.name;
+				var groupType = entry.type || 'user';
+
+				if (groupType === 'user' && !Config.defaultOrder) Config.defaultOrder = i; // this is where any undeclared groups will be positioned in userlist
+				if (!groupName) Config.defaultGroup = symbol;
+
+				Config.groups[symbol] = {
+					name: groupName ? Tools.escapeHTML(groupName + ' (' + symbol + ')') : null,
```

Comment:
Maybe i should store the data in a seperate object until the entire parsing is done, so any crashes wouldn't mess up the the entire client; it will just use the default ranks if the custom groups aren't completely loaded